controllers/auth.controller.ts

```diff
@@ -0,0 +1,43 @@
+import db from '../models/index'
+import { type IUser, type TypedRequestBody } from '../common/interfaces'
+import bcrypt from 'bcryptjs'
+import { type Response } from 'express'
+import jwt from 'jsonwebtoken'
+
+const User = db.models.User
+
+export const login: any = async (
+  req: TypedRequestBody<{
```

Comment:
looks like very generic typing, please add request validation and type request from class responsible for validation